RecipeApp.Core/RecipeConverter.cs

```diff
@@ -94,29 +102,59 @@ static double FindMultiplier(string cookingUnit)
             return -1;
         }
 
-        static void ConvertToCookingUnit(int index, string[] words)
+        private static string ConvertToCookingUnits(string recipe)
+        {
+            string[] words = recipe.Split(' ', '\n');
+
+            containsCookingOrSiUnits = false;
+
+            for (int index = 0; index < words.Length; index++)
+            {
+                StandardiseSiUnit(index, words);
+            }
+
+            if (!containsCookingOrSiUnits)
+            {
+                throw new InvalidRecipeException("No SI units in recipe.");
+            }
+
+            return string.Join(" ", words);
+        }
+
+        private static void StandardiseSiUnit(int index, string[] words)
```

Comment:
Instead, make this method return  a bool so you won't need to use a field but rather have it controlled in local scope. That confused me a bit.

Reply:
Have removed the field, made function return bool and moved the checking of different units into respective function.